src/Projection/PdoEventStoreProjector.php

```diff
@@ -603,7 +622,7 @@ private function fetchRemoteStatus(): ProjectionStatus
         return ProjectionStatus::byValue($result->status);
     }
 
-    private function handleStreamWithSingleHandler(MergedStreamIterator $events): void
+    private function handleStreamWithSingleHandler(MergedStreamIterator $events): bool
```

Comment:
It personnally would feel more easy to follow execution flow if gap detection would be communicated through an exception, instead of a boolean.

Reply:
Exceptions are more costly to build than simply returning a bool value. Also this is a private method, we are not communicating anything to the enduser.

Reply:
This :point_up: and in the run method the returned boolean is only checked if gap detection is enabled, otherwise it is completely ignored and has no effect.